src/ring_theory/derivation.lean

```diff
@@ -79,47 +84,62 @@ end
+            linear_map.to_fun_eq_coe, smul_algebra_smul_comm, coe_fn_coe, smul_add, add_comm],
+          ..(r • D : A →ₗ[R] M) }⟩
+
+@[simp] lemma coe_Rsmul (r : R) (D : derivation R A M) : ⇑(r • D) = r • D := rfl
+@[simp] lemma coe_Rsmul_linear_map (r : R) (D : derivation R A M) :
+  ↑(r • D) = (r • D : A →ₗ[R] M) := rfl
+lemma Rsmul_apply (r : R) (D : derivation R A M) : (r • D) a = r • D a := rfl
+
+instance has_scalar : has_scalar A (derivation R A M) :=
+⟨λ a D, { leibniz' := λ b c, by {
+            dsimp, simp only [smul_add, leibniz, smul_comm a, add_comm] },
+          ..(a • D : A →ₗ[R] M) }⟩
+
+@[simp] lemma coe_smul (a : A) (D : derivation R A M) : ⇑(a • D) = a • D := rfl
+@[simp] lemma coe_smul_linear_map (a : A) (D : derivation R A M) :
+  ↑(a • D) = (a • D : A →ₗ[R] M) := rfl
+lemma smul_apply (a : A) (D : derivation R A M) (b : A) : (a • D) b = a • D b := rfl
 
 instance : inhabited (derivation R A M) := ⟨0⟩
 
 instance : add_comm_monoid (derivation R A M) :=
-{ add := λ D1 D2, ⟨D1 + D2, λ a b, by { simp only [leibniz, linear_map.add_apply,
-    linear_map.to_fun_eq_coe, coe_fn_coe, smul_add], cc }⟩,
-  add_assoc := λ D E F, ext $ λ a, add_assoc _ _ _,
-  zero_add := λ D, ext $ λ a, zero_add _,
-  add_zero := λ D, ext $ λ a, add_zero _,
-  add_comm := λ D E, ext $ λ a, add_comm _ _,
-  ..derivation.has_zero }
+coe_injective.add_comm_monoid _ coe_zero coe_add
 
-@[simp] lemma add_apply : (D1 + D2) a = D1 a + D2 a := rfl
+/-- `coe_fn` as an `add_monoid_hom`. -/
+def coe_fn_add_monoid_hom : derivation R A M →+ (A → M) :=
+{ to_fun := coe_fn, map_zero' := coe_zero, map_add' := coe_add }
 
 @[priority 100]
 instance derivation.Rmodule : module R (derivation R A M) :=
-{ smul := λ r D, ⟨r • D, λ a b, by simp only [linear_map.smul_apply, leibniz,
-    linear_map.to_fun_eq_coe, smul_algebra_smul_comm, coe_fn_coe, smul_add, add_comm],⟩,
-  mul_smul := λ a1 a2 D, ext $ λ b, mul_smul _ _ _,
-  one_smul := λ D, ext $ λ b, one_smul _ _,
-  smul_add := λ a D1 D2, ext $ λ b, smul_add _ _ _,
-  smul_zero := λ a, ext $ λ b, smul_zero _,
-  add_smul := λ a1 a2 D, ext $ λ b, add_smul _ _ _,
-  zero_smul := λ D, ext $ λ b, zero_smul _ _ }
-
-@[simp] lemma smul_to_linear_map_coe : ↑(r • D) = (r • D : A →ₗ[R] M) := rfl
-@[simp] lemma Rsmul_apply : (r • D) a = r • D a := rfl
+function.injective.module R coe_fn_add_monoid_hom coe_injective coe_Rsmul
 
 instance : module A (derivation R A M) :=
-{ smul := λ a D, ⟨a • D, λ b c, by { dsimp, simp only [smul_add, leibniz, smul_comm a, add_comm] }⟩,
-  mul_smul := λ a1 a2 D, ext $ λ b, mul_smul _ _ _,
-  one_smul := λ D, ext $ λ b, one_smul A _,
-  smul_add := λ a D1 D2, ext $ λ b, smul_add _ _ _,
-  smul_zero := λ a, ext $ λ b, smul_zero _,
-  add_smul := λ a1 a2 D, ext $ λ b, add_smul _ _ _,
-  zero_smul := λ D, ext $ λ b, zero_smul A _ }
-
-@[simp] lemma smul_apply : (a • D) b = a • D b := rfl
+function.injective.module A coe_fn_add_monoid_hom coe_injective coe_smul
```

Comment:
These lines were the main intent of the PR - I figured I'd add all the missing `rfl` lemmas I could think of while golfing these, which ended up in the file actually being longer. Thankfully, it also means that `dsimp` is much more powerful now when using `derivation`s